### PR TITLE
WSL: Wait for moby socket proxy to stabilize

### LINKED ISSUE #4474: Importing RDX-proxy image fails on Windows w/ moby
### Actual Behavior

After a fresh installation of Windows 10 attempted to install the [RC](https://github.com/rancher-sandbox/rancher-desktop/actions/runs/4703529957), and it seems to be failing now with the following error:

Screen Shot 2023-04-17 at 11 27 14 AM

```
2023-04-17T18:14:47.163Z: Registered distributions: 
2023-04-17T18:14:51.448Z: Registered distributions: rancher-desktop
2023-04-17T18:14:51.562Z: Registered distributions: rancher-desktop
2023-04-17T18:14:51.563Z: Creating initial data distribution...
2023-04-17T18:14:55.464Z: Did not find a valid mount, mounting /mnt/wsl/rancher-desktop/run/data
2023-04-17T18:14:57.852Z: Installing C:\Program Files\Rancher Desktop\resources\resources\linux\internal\trivy as /mnt/c/Program Files/Rancher Desktop/resources/resources/linux/internal/trivy into /usr/local/bin/trivy ...
2023-04-17T18:14:58.205Z: Installing C:\Program Files\Rancher Desktop\resources\resources\linux\internal\rancher-desktop-guestagent as /mnt/c/Program Files/Rancher Desktop/resources/resources/linux/internal/rancher-desktop-guestagent into /usr/local/bin//rancher-desktop-guestagent ...
2023-04-17T18:15:01.432Z: WSL: executing: cat /root/.docker/config.json: Error: wsl.exe exited with code 1
```

Also, below is the error log from `extensions.log`
```
error during connect: this error may indicate that the docker daemon is not running: Post "http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.24/images/load?quiet=1": open //./pipe/docker_engine: The system cannot find the file specified.
```

### Steps to Reproduce

Install the following RC: https://github.com/rancher-sandbox/rancher-desktop/actions/runs/4703529957

### Result

RD fails to start.

### Expected Behavior

RD should start correctly. 

### Additional Information

Kubernetes was disabled. However, the following workaround works while backend set to moby:

- disable kubernetes
- shutdown RD
- wsl --unregister rancher-desktop
- wsl --unregister rancher-desktop-data
- wsl --shutdown
- and then start RD

### Rancher Desktop Version

https://github.com/rancher-sandbox/rancher-desktop/actions/runs/4703529957

### Rancher Desktop K8s Version

N/A

### Which container engine are you using?

Moby (Docker)

### What operating system are you using?

Windows

### Operating System / Build Version

Windows 10

### What CPU architecture are you using?

x64

### Linux only: what package format did you use to install Rancher Desktop?

None

### Windows User Only

No VPN, or Proxy

### PR DESCRIPTION
**Pull request by mook-as**
When we start the moby container engine, the socket proxy flaps a bit until it stabilizes.  Add extra checking to make sure it's stable before we allow the process to continue.

Fixes #4474

### CODE REVIEW

File: pkg/rancher-desktop/backend/wsl.ts [hunk 1]
```diff
@@ -1500,6 +1499,24 @@ export default class WSLBackend extends events.EventEmitter implements VMBackend
     ]);
   }
 
+  /**
+   * The docker proxy process flaps a bit at the start while things are starting
+   * up.  We need to wait for it to stabilize before declaring it ready.
```
Comment (Nino-K) [inline pkg/rancher-desktop/backend/wsl.ts:1504]: Can we also, elaborate on the retry strategy in the comment?

**Review by Nino-K — COMMENTED** [review]
Just a tiny comment, otherwise. LGTM

**Review by Nino-K — APPROVED** [review]

**Comment by mook-as** [comment]
@jandubois Do we need to cherry-pick this into the `release-1.9` branch, or some other branch?

**Comment by jandubois** [comment]
> @jandubois Do we need to cherry-pick this into the `release-1.9` branch, or some other branch?

Yes, into `release-1.9` please!

**Pull request merged**